lib/errors.js

```diff
@@ -0,0 +1,571 @@
+// Copyright (c) 2025 Mahmoud Alghalayini. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+/**
+ * Base error class for all rclnodejs errors.
+ * Provides structured error information with context.
+ * @class
+ */
+class RclNodeError extends Error {
+  /**
+   * @param {string} message - Human-readable error message
+   * @param {object} [options] - Additional error context
+   * @param {string} [options.code] - Machine-readable error code (e.g., 'TIMEOUT', 'INVALID_ARGUMENT')
+   * @param {string} [options.nodeName] - Name of the node where error occurred
+   * @param {string} [options.entityType] - Type of entity (publisher, subscription, client, etc.)
+   * @param {string} [options.entityName] - Name of the entity (topic name, service name, etc.)
+   * @param {Error} [options.cause] - Original error that caused this error
+   * @param {any} [options.details] - Additional error-specific details
+   */
+  constructor(message, options = {}) {
+    super(message);
+
+    // Maintains proper stack trace for where our error was thrown
+    Error.captureStackTrace(this, this.constructor);
+
+    this.name = this.constructor.name;
+    this.code = options.code || 'UNKNOWN_ERROR';
+    this.nodeName = options.nodeName;
+    this.entityType = options.entityType;
+    this.entityName = options.entityName;
+    this.details = options.details;
+
+    // Error chaining (ES2022 feature, Node.js 16.9+)
+    if (options.cause) {
```

Comment:
As rclnodejs requires minimum of nodejs `v16.13.0`

https://github.com/RobotWebTools/rclnodejs/blob/3ad842cc4b442ba18571b7b9d85e806dcd1ce4b3/package.json#L101

I think we can remove this `if` to simplify.